samples/web/content/apprtc/js/appcontroller.js

```diff
@@ -119,28 +127,69 @@ var AppController = function(loadingParams) {
     this.call_.onstatusmessage = this.displayStatus_.bind(this);
     this.call_.oncallerstarted = this.displaySharingInfo_.bind(this);
 
-    this.call_.start(this.loadingParams_.roomId);
-
-    window.onbeforeunload = this.call_.hangup.bind(this.call_);
-    document.onkeypress = this.onKeyPress_.bind(this);
-    window.onmousemove = this.showIcons_.bind(this);
-    
-    $(UI_CONSTANTS.muteAudioSvg).onclick = this.toggleAudioMute_.bind(this);
-    $(UI_CONSTANTS.muteVideoSvg).onclick = this.toggleVideoMute_.bind(this);
-    $(UI_CONSTANTS.fullscreenSvg).onclick = this.toggleFullScreen_.bind(this);
-    $(UI_CONSTANTS.hangupSvg).onclick = this.hangup_.bind(this);
-  
-    setUpFullScreen();
+    this.roomSelection_ = null;
+    if (this.loadingParams_.connect !== false && this.loadingParams_.roomId) {
+      // If we already have a room to connect to, connect now.
+      this.call_.start(this.loadingParams_.roomId);
+      
+      // Record this room in the recently used list.
+      var recentlyUsedList = new RoomSelection.RecentlyUsedList();
+      recentlyUsedList.pushRecentRoom(this.loadingParams_.roomId);
+      
+      this.finishCallSetup_();
+    } else {
+      // Display the room selection UI.
+      var roomSelectionDiv = $(UI_CONSTANTS.roomSelectionDiv);
+      this.roomSelection_ = new RoomSelection(roomSelectionDiv, this.loadingParams_.suggestedRoomId);
+      this.show_(roomSelectionDiv);
+      this.roomSelection_.onRoomSelected = function(roomName) {
+        this.hide_(roomSelectionDiv);
+        this.call_.start(roomName);
+        this.finishCallSetup_();
```

Comment:
I think it's cleaner to have all call setup done in one helper method, i.e.
if (should show landing page) {
  show landing page;
  onRoomSelected = {
      ...
      setupCall();
  }
  return;
}
setupCall();